control-plane-operator/controllers/hostedcontrolplane/cloud/powervs/powervs.go

```diff
@@ -0,0 +1,207 @@
+	config := map[string]string{
+		"ConfigFile":               fmt.Sprintf("%s/kubeconfig", kubeConfigFileBasePath),
+		"AccountID":                hcp.Spec.Platform.PowerVS.AccountID,
+		"ClusterID":                hcp.Name,
+		"G2workerServiceAccountID": hcp.Spec.Platform.PowerVS.AccountID,
+		"G2Credentials":            fmt.Sprintf("%s/%s", secretMountPath, "ibmcloud_api_key"),
+		"G2ResourceGroupName":      hcp.Spec.Platform.PowerVS.ResourceGroup,
+		"G2VpcSubnetNames":         hcp.Spec.Platform.PowerVS.VPC.Subnet,
+		"G2VpcName":                hcp.Spec.Platform.PowerVS.VPC.Name,
+		"Region":                   hcp.Spec.Platform.PowerVS.VPC.Region,
+		"PowerVSCloudInstanceID":   hcp.Spec.Platform.PowerVS.ServiceInstanceID,
+		"PowerVSRegion":            hcp.Spec.Platform.PowerVS.Region,
+		"PowerVSZone":              hcp.Spec.Platform.PowerVS.Zone,
+	}
+
+	configData := &bytes.Buffer{}
+	ccmConfigTemplate.Execute(configData, config)
+
+	if ccmConfig.Data == nil {
+		ccmConfig.Data = map[string]string{}
+	}
+
+	ccmConfig.Data[ccmConfig.Name] = configData.String()
+
+	return
+}
+
+func ReconcileCCMDeployment(deployment *appsv1.Deployment, hcp *hyperv1.HostedControlPlane, ccmConfig *corev1.ConfigMap, releaseImage *releaseinfo.ReleaseImage) (err error) {
+	commandToExec := []string{
+		"/bin/ibm-cloud-controller-manager",
+		"--authentication-skip-lookup",
+		"--bind-address=$(POD_IP_ADDRESS)",
+		"--use-service-account-credentials=true",
+		"--configure-cloud-routes=false",
+		"--cloud-provider=ibm",
+		fmt.Sprintf("--cloud-config=%s/%s", ccmConfigMapMountPath, ccmConfig.Name),
+		"--profiling=false",
+		"--leader-elect=true",
+		"--leader-elect-lease-duration=137s",
+		"--leader-elect-renew-deadline=107s",
+		"--leader-elect-retry-period=26s",
+		"--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_AES_128_GCM_SHA256,TLS_CHACHA20_POLY1305_SHA256,TLS_AES_256_GCM_SHA384",
+		fmt.Sprintf("--kubeconfig=%s/kubeconfig", kubeConfigFileBasePath),
+		"--use-service-account-credentials=false",
+	}
+
+	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas: &[]int32{1}[0],
+		Strategy: appsv1.DeploymentStrategy{Type: "Recreate"},
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"k8s-app": deployment.Name},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"k8s-app": deployment.Name},
+			},
+			Spec: corev1.PodSpec{
+				TerminationGracePeriodSeconds: &[]int64{90}[0],
+				Containers: []corev1.Container{
+					{
+						Name:            ccmContainerName,
+						Image:           releaseImage.ComponentImages()["powervs-cloud-controller-manager"],
+						ImagePullPolicy: "Always",
```

Comment:
please use if not present. Motivation is to be tolerant with registry outages and disconnected setups.

Reply:
sure, modified to `corev1.PullIfNotPresent`